sess.go

```diff
@@ -91,41 +92,42 @@ func (s *session) Serve() error {
 
 	// query-cycle
 	for {
-		msg, err = t.Read()
+		msg, err := t.NextFrontendMessage()
 		if err != nil {
 			return err
 		}
 
-		switch msg.Type() {
-		case protocol.Terminate:
+		switch msg.(type) {
```

Comment:
instead of explicit casting in each case, you can also do:
```suggestion
		switch v := msg.(type) {
```
or 
```suggestion
		switch msg := frontMsg.(type) {
```

Reply:
right! missed is :)